apps/prairielearn/elements/pl-string-input/pl-string-input.py

```diff
@@ -207,9 +215,21 @@ def parse(element_html: str, data: pl.QuestionData) -> None:
         data["submitted_answers"][name] = None
         return
 
+    # Do unicode decode
     if normalize_to_ascii:
-        a_sub = unidecode(a_sub)
-        data["submitted_answers"][name] = a_sub
+        a_sub = pl.full_unidecode(a_sub)
+
+    # Remove the leading and trailing characters
+    if remove_leading_trailing:
+        a_sub = a_sub.strip()
+
+    # Remove the blank spaces between characters
+    if remove_spaces:
+        a_sub = "".join(a_sub.split())
+
+    # Modify string case for submission and true answer to be lower.
+    if ignore_case:
+        a_sub = a_sub.lower()
```

Comment:
It's not entirely obvious to me that `ignore-case="true"` should actually mutate the submitted answer. It's a much more straightforward decision for the others where the very directly implies mutation ("normalize", "remove"). `ignore-case` feels like it should just control the grading algorithm. I'm willing to be convinced otherwise though!

Reply:
> Or add a different attribute entirely, yeah - that might be the best option. "ignore case" and "coerce to lowercase" have very different meanings to me, and the choice of lowercase vs. uppercase seems somewhat arbitrary. Maybe `transform-case="upper"`/`transform-case="lower"` so that we get an explicit mutation verb and so that they can control what actually happens?

I think that makes sense, we can build this out if people feel like they need it.

> All that said: it's not clear to me that this new attribute is needed. At this point, I would be inclined to switch `ignore-case` to _not_ mutate the submitted answer, and if someone complains about this behavior, we can add another attribute (in the spirit of not building things just for the sake of building them).

I'm in agreement, I'll go ahead and remove!

Reply:
@nwalters512 updated here: https://github.com/PrairieLearn/PrairieLearn/pull/8670/commits/4b64dd9925a6f86b270b5987c5356aa3e556ac17